### PR TITLE
RavenDB-6873 PR Review fixes

### PR DESCRIPTION
**Pull request by fitzchak**
https://github.com/ravendb/ravendb/pull/2312

### CODE REVIEW

File: src/Raven.Client/Documents/Commands/Batches/PutAttachmentCommandData.cs [hunk 1]
```diff
@@ -20,6 +20,23 @@ public PutAttachmentCommandData(string documentId, string name, Stream stream, s
...
+
+            if (Stream.CanSeek == false)
+                ThrowNotSeekableStream();
+            if (Stream.Position != 0)
+                ThrowPositionNotZero(Stream.Position);
+        }
+
+        private void ThrowPositionNotZero(long streamPosition)
+        {
+            throw new InvalidOperationException($"Cannot put an attachment with a stream that have position which isn't zero but {streamPosition}.");
```
Comment (ayende) [inline src/Raven.Client/Documents/Commands/Batches/PutAttachmentCommandData.cs:32]: error message strange "but 123" ?

Comment (ayende) [reply src/Raven.Client/Documents/Commands/Batches/PutAttachmentCommandData.cs:32]: Need to also explain _why_ we require this.

File: src/Raven.Server/Documents/Handlers/BatchHandler.cs [hunk 2]
```diff
@@ -90,16 +57,57 @@ public class BatchHandler : DatabaseRequestHandler
...
                 }
             }
         }
 
+        private async Task<MergedBatchCommand> CreateBatchCommand(DocumentsOperationContext context)
+        {
+            var command = new MergedBatchCommand {Database = Database};
+
+            if (HttpContext.Request.ContentType != null &&
+                HttpContext.Request.ContentType.StartsWith("multipart/mixed", StringComparison.OrdinalIgnoreCase))
```
Comment (ayende) [inline src/Raven.Server/Documents/Handlers/BatchHandler.cs:75]: Can it actually be anything _but_ "multipart/mixed" exactly?
The reason I'm asking is that I think that the browser will send it with `text/json` or some such, no?

Comment (ayende) [reply src/Raven.Server/Documents/Handlers/BatchHandler.cs:75]: And this is till a rare thing, so we want the common thing (just parsing the request to be the first thing here, then the request with attachment to be in a separate method.

File: src/Raven.Client/Documents/Commands/Batches/PutAttachmentCommandData.cs (hunk 1 above)
Comment (fitzchak) [reply src/Raven.Client/Documents/Commands/Batches/PutAttachmentCommandData.cs:32]: Fixed.

File: src/Raven.Server/Documents/Handlers/BatchHandler.cs (hunk 2 above)
Comment (fitzchak) [reply src/Raven.Server/Documents/Handlers/BatchHandler.cs:75]: Yes, the studio sends: 
Content-Type: application/json; charset=UTF-8

I will invert the check.

File: src/Raven.Client/Documents/Commands/Batches/PutAttachmentCommandData.cs [hunk 3]
```diff
@@ -20,6 +20,23 @@ public PutAttachmentCommandData(string documentId, string name, Stream stream, s
...
+        }
+
+        private void ThrowNotSeekableStream()
+        {
+            throw new InvalidOperationException(
+                "Cannot put an attachment with a not seekable stream. " +
+                "We require a seekable stream because we might failover to a differnt node if the current one is unavilalbe during the operation.");
```
Comment (arekpalinski) [inline src/Raven.Client/Documents/Commands/Batches/PutAttachmentCommandData.cs:39]: typos: unavilalbe -> unavailable, differnt -> different

Comment (fitzchak) [reply src/Raven.Client/Documents/Commands/Batches/PutAttachmentCommandData.cs:39]: Fixed, thanks!

**Pull request merged**